### ISSUE TITLE
Select2.js Searching both text and id

### ISSUE THREAD

**Issue by vuphuoc** [select2/select2#5335]
I'm using "Select2" plugin for my select to display all "Suppliers" from my table in SQL. I used Ajax to get all Suppliers and generated <option> tags with value and text for my select. I bound all <option> tags into my select and set "Select2" function for it. I have formatted the result to display two column like this:
![select2_asking](https://user-images.githubusercontent.com/41767525/43299787-fcd3d990-9185-11e8-9a20-4318af15034e.PNG)
I read the document of Select2 for formatting the searching. But I just can search the text (the second column in the picture above).

Here is the code I used for the result formatting and searching:
```
//Get NhaCungCap
function GetNhaCungCap() {
    ajaxFunc("/Services.aspx/GetAllNhaCungCap", { pageNumber: 0 }, function (data) {
        var json = JSON.parse(data.d);
        FillNhaCungCap(json);
    })
}
```
```
//Binding NhaCungCap
function FillNhaCungCap(json) {
    var html = "";
    $.each(json, function (k, v) {
        html += "<option value=" + v.MaNhaCungCap + ">" + v.TenNhaCungCap + "</option>";
    });

    $("#slNCC").html(html); //Bind <option> tag into select "slNCC"        
}
```
```
//Format slNCC using select2
 $('#slNCC').select2({
                placehoder: 'Tìm kiếm nhà cung cấp theo mã hoặc tên (F4)',
                escapeMarkup: function (markup) { return markup; }, // let our custom formatter work
                data: function (repo) {
                    return JSON.stringify({ q: repo.text || repo.id, page: params.page });// search term  
                },
               
                templateResult: formatRepo,
                templateSelection: formatRepoSelection
});  
```
```
//function to format Result
function formatRepo(repo) {
    if (repo.loading) {
        return  repo.id || repo.text;
    }
    var markup = "";

    markup += '<div class="row">';
    markup += '<div class="col-sm-4">' + repo.id + '</div>';
    markup += '<div class="col-sm-8">' + repo.text + '</div>';
    markup += '</div>';
    return markup;
}

```
```
function formatRepoSelection(repo) {
    return repo.full_name || repo.text;
}

```
Can you help me to solve this issue?
Thanks!

**Comment by pedrofurtado** [comment]
send us a jsfiddle example code, describing steps there. it is much more easy to understand and provide some solution. reopen when ready

**Comment by vuphuoc** [comment]
Thanks for your reply. In "data" property of ajax function, I used
"JSON.stringify()" function with my query. And then I did rename the
parameters from store procedure in my SQL Server to match with the
parameters I passed in data query. Finally, I could search data in bind it
to "select2".

Vũ Phước
FAQ and contact me here!
Gmail: *sgonline999@gmail.com <sgonline999@gmail.com>*
Phone: not available


Vào Th 3, 18 thg 9, 2018 vào lúc 00:07 Pedro Felipe de Azevedo Furtado <
notifications@github.com> đã viết:
